lib/instrumentation/generic-span.js

```diff
@@ -71,7 +71,7 @@ Object.defineProperty(GenericSpan.prototype, 'sampleRate', {
   enumerable: true,
   get () {
     const rate = parseFloat(this._context.tracestate.getValue('s'))
-    if (rate || rate === 0) {
+    if (rate >= 0 || rate <= 1) {
```

Comment:
```suggestion
    if (rate >= 0 && rate <= 1) {
```

You want `&&` here, I think.